packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -563,11 +564,26 @@ export class DataStores implements IDisposable {
 		await Promise.all(
 			Array.from(this.contexts)
 				.filter(([_, context]) => {
-					// Summarizer works only with clients with no local changes!
-					assert(
-						context.attachState !== AttachState.Attaching,
-						0x165 /* "Summarizer cannot work if client has local changes" */,
-					);
+					// Summarizer works only with clients with no local changes. A data store in attaching
+					// state indicates an op was sent to attach a local data store.
+					if (context.attachState === AttachState.Attaching) {
```

Comment:
Can you please double check that we get here only in summarizer client? I.e. it's not hit in various container serialization or on-demand summary cases.